cpp/src/arrow/compute/kernels/vector_selection_test.cc

```diff
@@ -42,99 +42,161 @@ using std::string_view;
 
 namespace compute {
 
+namespace {
+
+template <typename T>
+Result<std::shared_ptr<Array>> REEncode(const T& array) {
+  ARROW_ASSIGN_OR_RAISE(auto datum, RunEndEncode(array));
+  return datum.make_array();
+}
+
+Result<std::shared_ptr<Array>> REEFromJson(const std::shared_ptr<DataType>& ree_type,
```

Comment:
Nit: use same casing as existing helpers?
```suggestion
Result<std::shared_ptr<Array>> REEFromJSON(const std::shared_ptr<DataType>& ree_type,
```